tests/realtikvtest/pessimistictest/pessimistic_test.go

```diff
@@ -3381,3 +3384,162 @@ func TestLazyUniquenessCheckWithStatementRetry(t *testing.T) {
 	require.False(t, tk.Session().GetSessionVars().InTxn())
 	tk.MustExec("admin check table t5")
 }
+
+func TestRCPointWriteLockIfExists(t *testing.T) {
```

Comment:
Consider covering the cases that the key is already locked in a previous statement of the transaction, including both existing or not-existing cases.

Reply:
I have added the case that  key is already locked in a previous statement when the key exists.
I'll try to add the case that key is already locked in a previous statement but the key doesn't exists.